### ISSUE TITLE
Improve UX on context selector

### ISSUE THREAD

**Issue by goenning** [aptakube/aptakube#17]
### What problem are you trying to solve?

Some users have reported that current context selector is not optimal with hundreds of contexts.

### Describe the solution you'd like

On v0.7.0 we're adding a search functionality and this issue #13 might help a bit as well. But a few others things people suggested:

- Create groups of contexts, so users can select a group instead of selecting multiple individual contexts;
- Mark contexts and groups as favorites: make it easier to find them;
- Double click to connect to a set: to avoid having to select the context and then click on the "Continue" button. I'm not too sure about this one, might be a bit confusing.

**Comment by gimbalgambit** [comment]
Hi! It'd be great to support adding an alias to the contexts that are loaded, so that people who have to use specific, very long `context.name` values in their kubeconfigs for other CLI tooling reasons can give those contexts a nickname or something shorter or more human-readable!

**Comment by goenning** [comment]
Thanks @gimbalgambit for the suggestion, that's been requested a few times as well, so I'll probably add an option for alias (which only applies to the UI, it won't touch your kubeconfigs). One other suggestion from the community was to add a toggle between grid/list view. The list view would work better for longer names, what you think?

**Comment by gimbalgambit** [comment]
@goenning Thanks for your response! I think the List/Grid toggle would be good... would definitely help with longer names. In our use-case, we have context names that are AWS ARNs, so they're pretty hard to differentiate even if you can read the whole thing (e.g. `are:aws:eks:us-east-1:1122334455:cluster/whatever` is hard to find next to `are:aws:eks:us-east-1:1122334855:cluster/whatever`).

**Comment by goenning** [comment]
Great, next version will have a list view so that should help you until alias is supported. 

AWS should just stop using ARN on context names 😅

**Comment by gimbalgambit** [comment]
Thank you! Appreciate your work. 🙇

**Comment by goenning** [comment]
List View is now available on 1.1.0, but I'll keep this issue open as there is a few more improvements we can do around context selector.

**Comment by elmariofredo** [comment]
It would be great if I can switch between contexts from anywhere without need to go to list of contexts first, e.g. popup selector when clicking on context name? Also regarding switching, now seems that app will not open same view ( eg. same pod detail ) as last opened, so IMHO context switching is bit more painful than it should. Thanks

**Comment by goenning** [comment]
> It would be great if I can switch between contexts from anywhere without need to go to list of contexts first, e.g. popup selector when clicking on context name?

Yeah, we could improve that too. I've updated the original description to mention this and not miss it.

> Also regarding switching, now seems that app will not open same view ( eg. same pod detail ) as last opened, so IMHO context switching is bit more painful than it should. Thanks

Can you expand on it? I didn't quite understand what the problem is.

**Comment by MattKetmo** [comment]
IMO the command palette is the perfect candidate to cover this use case.
Cmd-K to switch context, or go to the Pod list page, or go to a specific pod, etc…

Here is a quick illustration of what I've in mind:
![image](https://user-images.githubusercontent.com/334996/218163890-a0d3e4f7-fac2-4f22-8092-b5dae6fef0b3.png)

**Comment by elmariofredo** [comment]
> > Also regarding switching, now seems that app will not open same view ( eg. same pod detail ) as last opened, so IMHO context switching is bit more painful than it should. Thanks
> 
> Can you expand on it? I didn't quite understand what the problem is.

Sure :)

1. You will open pod detail in cluster-prod 
2. Realise that you want to compare it with pod detail in cluster-test
3. You will open cluster-test and get into desired pod detail
4. Then you will try to open cluster-prod again and you are no longer on pod detail but on Deployments which are default view

It would be great if Aptakube remembers last view of specified cluster

> IMO the command palette is the perfect candidate to cover this use case.
> Cmd-K to switch context, or go to the Pod list page, or go to a specific pod, etc…

yes! that look like great way to switch clusters.

**Comment by goenning** [comment]
@MattKetmo that's really good, I like it. That's how I envisioned the Command Bar being built, although I didn't think of context switching in it. Thanks for sharing it.

Just one other thing we'd need to consider here: Switching to multiple contexts. Maybe this would be a good fit with the "Context Set", where a user can group various contexts into a set and then use the command bar to switch to a single context or to a context set

**Comment by aexvir** [comment]
this is also my main frustration with aptakube at the moment, it's cumbersome to navigate between resources and I feel that keyboard shortcuts are not a first class citizen here, as most of the ui is better navigated with the mouse

I think that the command palette suggestion is ideal, and I'd love to see it implemented \
I'm very used to navigating exclusively with the keyboard in `k9s` and I'm much slower with aptakube at the moment

**Comment by goenning** [comment]
Release 1.3.0 is coming out (hopefully!) this week. So I spent sometime today thinking about what I would like to add to 1.4.0, and I think this next release will be dedicated to UX/UI, with a big focus on command palette and various other keyboards shortcuts, as well as some other quality of life improvements!

**Comment by goenning** [comment]
Keyboard navigation on context selector is now available on 1.5, but I'll keep this issue open as there's more work to do!

**Comment by oscar-b** [comment]
@goenning It would be nice to have some way to quickly see if you're in for instance `prod` or `test` context (tags?), maybe using colors somewhere (header, sidebar?) based on those tags.
